### PR TITLE
Can register using an email address via identity server

### PR DESCRIPTION
**Pull request by PiotrKozimor**
New test case. Why is it neccessary? Test case `Can register using an email address` expect homeserver to send email. There is currently no way to test configuration when identity server is requested to send email (by [/requestToken](https://spec.matrix.org/unstable/identity-service-api/#post_matrixidentityapiv1validateemailrequesttoken) endpoint) to mocked identity server. When received, email is considered verified (the part with [/submitToken](https://spec.matrix.org/unstable/identity-service-api/#get_matrixidentityapiv1validateemailsubmittoken) is skipped).

There is backward-compatible change in configuration provided to Dendrite. Is is described in [PR](https://github.com/matrix-org/dendrite/pull/1837) for Dendrite

Signed-off-by: `Piotr Kozimor <p1996k@gmail.com>`

### CODE REVIEW

File: tests/11register.pl
Comment (danpegk) [inline tests/11register.pl:733]: @PiotrKozimor Remove space? 😅

**Review by danpegk — CHANGES_REQUESTED** [review]

File: tests/11register.pl
Comment (PiotrKozimor) [reply tests/11register.pl]: Done

**Comment by PiotrKozimor** [comment]
@richvdh how do I request review on this PR?

**Comment by clokep** [comment]
> how do I request review on this PR?

I've set he review flag for you! 😄

File: run-tests.pl
Comment (erikjohnston) [inline run-tests.pl:57]: Why do we now need to set an explicit port ooi?

File: tests/11register.pl
Comment (erikjohnston) [inline tests/11register.pl:762]:
```suggestion
```

File: tests/11register.pl
Comment (erikjohnston) [inline tests/11register.pl:779]:
```suggestion
         log_if_fail "First /register body", $body;
```

Since we may fail for other reasons.

**Review by erikjohnston — COMMENTED** [review]
I'm failing to remember quite how the registration stuff works, so I'll leave for someone else to review properly.

**Comment by erikjohnston** [comment]
Is this blocked until https://github.com/matrix-org/dendrite/pull/1837 lands?

File: run-tests.pl
Comment (PiotrKozimor) [reply run-tests.pl:57]: So that it can be added to trusted servers in homeserver configuration. I suppose that each time `id_server_fixture` is called random port is assigned.

**Comment by PiotrKozimor** [comment]
> Is this blocked until [matrix-org/dendrite#1837](https://github.com/matrix-org/dendrite/pull/1837) lands?

Good question, I think this PR and the one mentioned by you blocks each other. I can think of following way to resolve this:
* Overwrite dendrite branch in sytest CI,
* Merge this PR once all checks are successful,
* Merge dendrite PR with check run against default sytest branch. 

BTW this test case is not whitelisted in dendrite nor synapse for now - only in [this branch](https://github.com/globekeeper/dendrite/tree/piotrkozimor/registering-with-an-email-support)

File: run-tests.pl
Comment (erikjohnston) [reply run-tests.pl:57]: Oh yes, I'm blind. We don't need it for Synapse because there's an option to trust all ID servers, right?

Comment (PiotrKozimor) [reply run-tests.pl:57]: It is quite messy in Synapse. For example in `/register/email/requestToken` `id_server` field [will be just ignored](https://github.com/matrix-org/synapse/blob/develop/synapse/rest/client/v2_alpha/register.py#L150). I could not figure out how does it work in user-interactive authentication. I didn't either found option to trust all ID servers in Synapse.

**Comment by PiotrKozimor** [comment]
We want to stick with HS sending validation emails - see [this issue](https://github.com/matrix-org/synapse/issues/1710), so closing this PR.

**Pull request closed without merging**
